src/validators/required.ts

```diff
@@ -0,0 +1,52 @@
+import { FieldValidationFunctionSync, ValidationResult } from '../model';
+
+let errorMessage = 'Please fill in this mandatory field.';
+
+export const VALIDATOR_TYPE = 'REQUIRED';
+
+export const setErrorMessage = (message: string) => {
+  errorMessage = message;
+};
+
+export interface RequiredArgs {
+  trim: boolean;
+}
+
+const DEFAULT_PARAMS: RequiredArgs = { trim: true };
+
+const createDefaultValidationResult = (): ValidationResult => ({
+  succeeded: true,
+  type: VALIDATOR_TYPE,
+  message: '',
+});
+
+const isStringValid = (value: string, trim: boolean): boolean => {
+  return trim ? value.trim().length > 0 : value.length > 0;
+};
+
+const isValidField = (value: any, trim: boolean): boolean => {
+  return typeof value === 'string'
+    ? isStringValid(value, trim)
+    : value === true || typeof value === 'number';
+};
+
+export const required: FieldValidationFunctionSync = (
+  value,
+  values,
+  customArgs: RequiredArgs = DEFAULT_PARAMS,
+  customMessage
+) => {
+  if (!customArgs) {
```

Comment:
This `if` is to cover `null` case?

Reply:
yes, the case is: I want to enter a customMessage but I don't want to inform customArgs I just pass null.

That's where I think if an structure could be a good idea (again cool refactor :P) what's your take on this @nasdan?